pkg/cloudevents/transport/http/options.go

```diff
@@ -145,7 +146,18 @@ func WithStructuredEncoding() Option {
 	}
 }
 
-// WithPort sets the port for for clients with HTTP transports.
+func (t *Transport) checkListen(prefix string) error {
```

Comment:
I would expect this method to be:

```golang
func checkListen(t *Transport, prefix string) error
```

The reason these option functions exist in this location is to gain access to private members of the http transport struct.

Reply:
There are no private members in Go, only exported and not exported. Nothing is private within a package.
I've no objection to changing it as a matter of style.

Reply:
done